Python/PyPlasma.cpp

```diff
@@ -60,3 +60,23 @@ int PyType_CheckAndReady(PyTypeObject* type)
         ST::printf(stderr, "WARN: Failed to ready {}\n", type->tp_name);
     return result;
 }
+
+#if (PY_MAJOR_VERSION < 3) || ((PY_MAJOR_VERSION == 3) && (PY_MINOR_VERSION < 2))
+int PyUnicode_FSDecoder(PyObject* value, void* result)
+{
+    if (!value || value == Py_None)
+        return 0;
+
+    // This will rely on a future call to PyAnyString_AsSTString to properly decode the path
+    if (PyAnyString_Check(value)) {
+        *((PyObject**)result) = value;
+        Py_INCREF(value);
+
+        // Py_CLEANUP_SUPPORTED was not added until Python 3.1, so we will rely on the caller
+        // to clean up the reference.
+        return 1;
+    }
+
+    return 0;
+}
```

Comment:
Rather than try to understand the compatibility differences between this and the versions in Python 3.2+ (hint, it has changed a fair bit even since 3.2), why not provide a new converter which does the conversion directly to `ST::string` and handles `os.PathLike` appropriately for Python 3.6+?

Reply:
That seems fair. Would you prefer calling the `__fspath__()` method or continuing to use `PyUnicode_FSDecoder()`? The benefit to the latter is the ability to use `bytes` paths... I'm not sure why that is desirable, but it does give us that option.

Reply:
I'm fine with using `PyUnicode_FSDecoder` when it's available.

Reply:
This function is no longer needed with your updates.